go/vt/vttablet/onlineddl/executor.go

```diff
@@ -3098,6 +3098,20 @@ func shouldCutOverAccordingToBackoff(
 	return false, false
 }
 
+// getDependentMigrations returns a slice of migrations that must cut-over in-order, before the provided
+// migration.
+func getDependentMigrations(onlineDDL *schema.OnlineDDL, pendingMigrationsUUIDs []string) []string {
+	dependentMigrations := make([]string, 0, len(pendingMigrationsUUIDs))
+	for _, pendingMigrationsUUID := range pendingMigrationsUUIDs {
+		if pendingMigrationsUUID == onlineDDL.UUID {
+			// found all dependencies if we found ourself
+			break
+		}
+		dependentMigrations = append(dependentMigrations, pendingMigrationsUUID)
+	}
```

Comment:
So the order is guaranteed, with the final one being ourselves? If so, shouldn't we return a nil/empty slice if the length is < 2 and return a sub-slice, eliding the last element?

Reply:
@mattlord this is an ordered list of migrations, with the final one NOT being guaranteed to be ourselves. It is possible for migrations to be queued after ours

We can only guaranteed the migrations "ahead" of us are dependencies when using `--in-order-completion`. cc @shlomi-noach to validate my understanding